mrp_unbuild_rebuild_variant/tests/test_mrp_unbuild_rebuild_variant.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import exceptions
+from odoo.tests import SavepointCase
+
+
+class TestMrpUnbuildRebuildVariant(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+        ref = cls.env.ref
+
+        # Ref products
+        cls.product_amplifier_phono_tuner = ref(
+            "mrp_unbuild_rebuild_variant."
+            "product_product_amplifier_with_phono_with_tuner"
+        )
+        cls.product_amplifier_tuner = ref(
+            "mrp_unbuild_rebuild_variant."
+            "product_product_amplifier_without_phono_with_tuner"
+        )
+        cls.product_amplifier_phono = ref(
+            "mrp_unbuild_rebuild_variant."
+            "product_product_amplifier_with_phono_without_tuner"
+        )
+        cls.product_amp_case = ref(
+            "mrp_unbuild_rebuild_variant.product_product_amplifier_case"
+        )
+        cls.product_amp_preamp = ref(
+            "mrp_unbuild_rebuild_variant.product_product_amplifier_preamp"
+        )
+        cls.product_amp_power_amp = ref(
+            "mrp_unbuild_rebuild_variant.product_product_amplifier_power_amp"
+        )
+        cls.product_amp_phono = ref(
+            "mrp_unbuild_rebuild_variant.product_product_amplifier_phono"
+        )
+        cls.product_amp_tuner = ref(
+            "mrp_unbuild_rebuild_variant.product_product_amplifier_tuner"
+        )
+
+        # Ref bom, locations
+        cls.amplifier_bom = ref("mrp_unbuild_rebuild_variant.mrp_bom_amplifier")
+        cls.stock_location = ref("stock.stock_location_stock")
+
+        # Init inventory
+        inventory_matrix = {
+            cls.product_amp_case: ["000123", ],
+            cls.product_amp_preamp: ["100456", ],
+            cls.product_amp_power_amp: ["200789", ],
+            cls.product_amp_phono: ["300321", ],
+            cls.product_amp_tuner: ["400654", ],
+        }
```

Comment:
Would be better to keep different lot numbers in stock to ensure the components used in the rebuild are the ones generated by the unbuild order and not those that are in stock.

Reply:
Yep, I create them when I need them.
It makes thins easier, when I try to test the case where there's not enough stock until an amplifier is dismantled.